Tasks/VsTest/Helpers.ps1

```diff
@@ -133,3 +133,36 @@ function Locate-VSVersion()
 	}
 	return $version
 }
+
+function GetResultsLocation {
+	[cmdletbinding()]
+	[OutputType([System.String])]
+	param(		
+		[string]$runSettingsFilePath		
+	)
+
+    # If this is a runsettings file then try to get the custom results location from it
+    
+    if((([string]::Compare([io.path]::GetExtension($runSettingsFilePath), ".runsettings", $True) -eq 0) -Or ([string]::Compare([io.path]::GetExtension($runSettingsFilePath), ".tmp", $True) -eq 0)) -And 
+    !(Test-Path $runSettingsFilePath -pathtype container))
+    {
+        $runSettingsForTestResults = [System.Xml.XmlDocument](Get-Content $runSettingsFilePath)
```

Comment:
If the node we are looking for is not there, it will default to old behaviour. If the xml structure itself is wrong, it will fail as expected with appropriate error.